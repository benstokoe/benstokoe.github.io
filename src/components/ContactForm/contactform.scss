@import '../../scss/spacing';
@import '../../scss/breakpoints';
@import '../../scss/colours';

.contact-form {
  &__submit {
    display: block;
    width: 100%;
    max-width: 100%;
    background-color: $dark;
    color: white;
    padding: spacing(2) 0;

    @include breakpoint($desktop) {
      margin: 0;
    }
  }
}
