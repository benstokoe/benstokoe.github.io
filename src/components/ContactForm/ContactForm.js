import React, { useState } from 'react';
import { Formik } from 'formik';
import * as Yup from 'yup';
import { camelCase } from 'lodash';
import Input from './input/Input';
import './contactform.scss';

const ContactForm = () => {
  const [submitted, setSubmitted] = useState(false);

  const validationSchema = Yup.object().shape({
    fullName: Yup.string().required('Please enter your name.'),
    emailAddress: Yup.string().required('Please enter your email.'),
    message: Yup.string().required('Please enter your message.'),
  });

  const handleSubmit = (values, { setSubmitting }) => {
    console.log('handleSubmit', values);

    // TODO: do something with the message.

    setSubmitted(true);

    setSubmitting(false);
  };

  const fields = [
    { label: 'Full name' },
    { label: 'Email address' },
    { label: 'Message', type: 'textarea' },
  ];

  return (
    <div className="contact-form">
      {!submitted ? (
        <Formik
          onSubmit={handleSubmit}
          validationSchema={validationSchema}
          render={({
            values,
            handleSubmit,
            errors,
            handleChange,
            handleBlur,
            isSubmitting,
          }) => (
            <form onSubmit={handleSubmit}>
              {fields.map(field => {
                const id = camelCase(field.label);

                return (
                  <Input
                    key={id}
                    onChange={handleChange}
                    onBlur={handleBlur}
                    id={id}
                    name={id}
                    type={field.type}
                    defaultValue={values[id]}
                    placeholder={field.label}
                    error={errors[id]}
                  />
                );
              })}

              <button
                type="submit"
                className="contact-form__submit block-link block-link--chevron"
                disabled={isSubmitting}
              >
                Send it!
              </button>
            </form>
          )}
        />
      ) : (
        <p>
          Thank you for contacting me. I will aim to get back to you within 48
          hours.
        </p>
      )}
    </div>
  );
};

export default ContactForm;
