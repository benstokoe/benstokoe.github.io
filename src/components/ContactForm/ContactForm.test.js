import React from 'react';
import { shallow } from 'enzyme';
import ContactForm from './ContactForm';

describe('ContactForm', () => {
  let wrapper;
  const props = {};

  beforeEach(() => {
    wrapper = shallow(<ContactForm {...props} />);
  });

  it('should render properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
