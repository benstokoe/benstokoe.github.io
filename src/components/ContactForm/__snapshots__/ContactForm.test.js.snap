// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactForm should render properly 1`] = `
<div
  className="contact-form"
>
  <Formik
    enableReinitialize={false}
    isInitialValid={false}
    onSubmit={[Function]}
    render={[Function]}
    validateOnBlur={true}
    validateOnChange={true}
    validationSchema={
      ObjectSchema {
        "_blacklist": RefSet {
          "list": Set {},
          "refs": Map {},
        },
        "_conditions": Array [],
        "_defaultDefault": [Function],
        "_deps": Array [],
        "_excludedEdges": Array [],
        "_exclusive": Object {},
        "_mutate": false,
        "_nodes": Array [
          "message",
          "emailAddress",
          "fullName",
        ],
        "_options": Object {
          "abortEarly": true,
          "recursive": true,
        },
        "_type": "object",
        "_typeError": [Function],
        "_whitelist": RefSet {
          "list": Set {},
          "refs": Map {},
        },
        "fields": Object {
          "emailAddress": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "required": false,
            },
            "_mutate": false,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
          },
          "fullName": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "required": false,
            },
            "_mutate": false,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
          },
          "message": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "required": false,
            },
            "_mutate": false,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
          },
        },
        "tests": Array [],
        "transforms": Array [
          [Function],
        ],
      }
    }
  />
</div>
`;
